Model/Api/ShippingMethods.php

```diff
@@ -779,7 +779,9 @@ public function getShippingOptions($quote, $addressData)
                 ->setTaxAmount($taxAmount);
         }
 
-        $shippingAddress->setShippingMethod(null)->save();
+        $shippingAddress->setShippingMethod(null);
+        $this->totalsCollector->collectAddressTotals($quote, $shippingAddress);
```

Comment:
Could you please check how much time adds this additional call in a typical case?

Reply:
@vitaliyreznikov, @danac-gs, @matt-thomason, @njfaries 

On average, out of 10 shipping hooks, the totals collection adds 30 ms
More details here: https://docs.google.com/spreadsheets/d/10z7-RwuyxfEVSGmAVK5D0ucW-kL66lbWxEcQFdWY5sk/edit?usp=sharing